### DISCUSSION TITLE
Is it possible to know the max width of a section?

### DISCUSSION THREAD

**Discussion by milanglacier** [nvim-lualine/lualine.nvim#664] [Q&A]
Currently I have a component which will be a long string. I will make truncation on that string since I don't want the string to be too lengthy to occupy the entire section (I place it at section C with other components), I want to dynamically control the length of the string.
Currently I dynamically control the length of that string using `nvim_get_win_width()`,
so my current logic will be

```
win_width = vim.api.nvim_get_win_width()
size_of_my_string = win_width * 0.6
```

But this is a less flexible way: since the `win_width` is dynamically changed, if the `win_width` is large, then `win_width * 0.6` would be truncated "too much", there are still many left space can be utilized.
But if `win_width` is too small, then `win_width * 0.6` would be too lengthy, the other components at `section C` will be undisplayed since `size_of_my_string` will occupy the entire space of `section_c`.

Is is possible to know the max width of `section c` and make dynamically change of size of string based on that?

**Comment by shadmansaleh** [comment]
I don't think there's a way to do that. At least none that I know of.
